### PR TITLE
fix: reliably exit plan mode on Implement this plan

### LINKED ISSUE #462: Plan mode: "Implement this plan" can stay in plan mode and stall turn startup
## Problem
When a user clicks **Implement this plan** from the plan follow-up card, the next turn can fail in two related ways:

1. **Functional mode-switch failure**: the next turn still runs with plan-mode behavior (or appears to stall after the initial goal block) instead of switching to implementation mode.
2. **UI/state-sync failure**: the collaboration mode UI and thread-scoped codex params can remain on `plan`, causing future turns to keep inheriting plan mode unexpectedly.

## Why this is happening
Current plan-accept flow depends on collaboration modes being loaded and resolvable at click time. In race/fallback scenarios (e.g. mode list not yet loaded, stale selection, or only `plan` available in-memory), the follow-up message can be sent without a safe non-plan override and without persistently updating thread-scoped mode state.

## Reproduction
1. Start a thread in plan mode and wait for a plan tool output.
2. Click **Implement this plan** quickly after reconnect/thread switch (or when mode list is still refreshing).
3. Observe one or more outcomes:
   - turn starts with plan-like behavior instead of implementation behavior,
   - agent emits initial goal block then appears stuck/stopped,
   - collaboration mode UI still shows/sticks to `plan` for that thread.

## Expected behavior
- Clicking **Implement this plan** should deterministically start the next turn in a non-plan mode.
- UI selection and persisted thread codex params should switch out of `plan` in lockstep.
- If no safe non-plan mode is currently resolvable, the follow-up turn should still send an explicit neutral mode override rather than inheriting stale `plan` state.

## Proposed fix (frontend-focused)
- Harden mode resolution for plan-accept:
  - prefer `default`, then `code`, then first non-`plan` mode,
  - never fallback to `plan` for implement action.
- Always send explicit per-turn collaboration-mode option on plan-accept:
  - resolved non-plan payload when available,
  - explicit `collaborationMode: null` when not.
- Persist thread-scoped collaboration mode id during plan follow-up actions so UI + behavior stay aligned.

## Acceptance criteria
- [ ] `Implement this plan` never sends a plan-mode payload.
- [ ] In plan-only/empty-mode race states, accept sends explicit neutral override (`collaborationMode: null`).
- [ ] Thread codex params persist collaboration mode transition on accept (non-plan or null).
- [ ] Visual collaboration mode state remains consistent after accept, including thread switch/reselect.
- [ ] Regression tests cover default/code/non-plan fallback and plan-only/empty-mode race scenarios.

### PR DESCRIPTION
**Pull request by amanthanvi**
## Summary
- fix plan-followup accept flow so `Implement this plan` never falls back to `plan` mode
- keep collaboration mode UI/state and thread-scoped persisted codex params in sync when accepting/submitting plan follow-ups
- add regression tests for default/code/non-plan fallback and plan-only race behavior

## Problem
After clicking `Implement this plan`, some turns could start with stale plan-mode behavior (or appear to stall after the initial goal block), and UI mode state could remain on `plan` for the thread.

## Root cause
Plan-accept mode selection could race with collaboration mode availability and fallback to an unsafe selection path, while thread-scoped mode persistence was not updated during plan follow-up actions.

### `src/features/app/hooks/usePlanReadyActions.ts`
- added deterministic implementation-mode resolver:
  - `default` -> `code` -> first non-`plan` -> none
- for plan accept:
  - updates selected collaboration mode id to resolved non-plan id (or `null`)
  - persists thread codex `collaborationModeId` to same value
  - always sends explicit turn override (`collaborationMode` payload or `null`)
- for plan changes submit:
  - persists plan mode id (or `null`) alongside selection
  - always sends explicit turn override (`collaborationMode` payload or `null`)

### `src/App.tsx`
- wired `persistThreadCodexParams` into `usePlanReadyActions`

### `src/features/app/hooks/usePlanReadyActions.test.tsx`
- new regression suite covering:
  - default-mode accept path
  - non-plan fallback when default/code absent
  - plan-only fallback to explicit neutral override
  - disconnected workspace reconnect path
  - plan changes submit persistence behavior

## Linked issue
- Closes #462

## Validation
Executed before branch transfer (same code/commit content):
- `npm run lint`
- `npm run test`
- `npm run typecheck`
- `cd src-tauri && cargo check`
- targeted:
  - `npm run test -- src/features/app/hooks/usePlanReadyActions.test.tsx`
  - `npm run test -- src/features/messages/components/Messages.test.tsx src/features/collaboration/hooks/useCollaborationModes.test.tsx`

Note: this branch was moved into an isolated worktree to avoid interfering with concurrent work on `feat/followup-queue-steer-460`.

### CODE REVIEW

File: src/features/app/hooks/usePlanReadyActions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    });
+    expect(sendUserMessageToThread).toHaveBeenCalledWith(
+      connectedWorkspace,
+      "thread-1",
+      "[[cm_plan_ready:accept]] Implement this plan.",
+      [],
+      { collaborationMode: null },
+    );
+  });
+
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply src/features/app/hooks/usePlanReadyActions.test.tsx:158]: Implemented in 9d5c8e7. Added explicit coverage for an empty collaboration mode list in handlePlanAccept, asserting neutral override (collaborationMode: null) and persisted collaborationModeId: null.

File: src/features/app/hooks/usePlanReadyActions.test.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+          mode: "plan",
+          settings: {
+            developer_instructions: null,
+            model: "gpt-5.2",
+            reasoning_effort: "high",
+          },
+        },
+      },
+    );
+  });
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply src/features/app/hooks/usePlanReadyActions.test.tsx:234]: Implemented in 9d5c8e7 with a safety adjustment: when handlePlanSubmitChanges cannot find plan mode, we now keep selected/persisted mode untouched and still send explicit neutral override (collaborationMode: null). Added regression test coverage for this path.

File: src/features/app/hooks/usePlanReadyActions.ts [hunk 3]
```diff
@@ -134,17 +155,19 @@ export function usePlanReadyActions({
       }
 
       const planMode = findCollaborationMode("plan");
-      if (planMode?.id) {
-        setSelectedCollaborationModeId(planMode.id);
-      }
+      const planModeId = planMode?.id ?? null;
+      setSelectedCollaborationModeId(planModeId);
+      persistThreadCodexParams({
+        collaborationModeId: planModeId,
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply src/features/app/hooks/usePlanReadyActions.ts]: Implemented in 9d5c8e7. handlePlanSubmitChanges now updates UI + persisted mode only when plan mode exists. When modes are transiently empty, it no longer clears stored selection and sends follow-up with explicit neutral override (collaborationMode: null). Added regression test for this edge case.

**Comment by amanthanvi** [comment]
@codex review

File: src/features/app/hooks/usePlanReadyActions.test.tsx [hunk 4]
```diff
@@ -0,0 +1,260 @@
+// @vitest-environment jsdom
+import { act, renderHook } from "@testing-library/react";
+import { describe, expect, it, vi } from "vitest";
+import type { CollaborationModeOption, WorkspaceInfo } from "../../../types";
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply src/features/app/hooks/usePlanReadyActions.test.tsx]: Implemented in f8c189b. Updated the new test module to use frontend aliases per repo contract: `@/types` and `@app/hooks/usePlanReadyActions`. Validation run: `npm run test -- src/features/app/hooks/usePlanReadyActions.test.tsx` and `npm run typecheck` both passed.

**Comment by Dimillian** [comment]
@codex review
